src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -20,17 +21,50 @@ export default class SimidMapCreative extends BaseSimidCreative {
   }
 
   /** @override */
-  onStart(eventData){
-    super.onStart(eventData);
-    //ToDo(juliareichel@): handle invalid JSON and param errors
-    const adParams = JSON.parse(this.creativeData.adParameters);
+  onInit(eventData) {
+    this.validateAndParseAdParams_(eventData);
+    super.updateInternalOnInit(eventData);
```

Comment:
Since SimidMapCreative does not define a updateInternalOnInit function, you don't need super here, you can use `this.updateInternal..`

Reply:
good to know!